torchgen/gen.py

```diff
@@ -210,7 +211,9 @@ def parse_native_yaml_struct(
             use_out_as_primary=True,
             external=False,
             # Only cuda-like devices in tree require device guards
-            device_guard=is_cuda_dispatch_key(k) or is_xpu_dispatch_key(k),
+            device_guard=is_cuda_dispatch_key(k)
+            or is_xpu_dispatch_key(k)
+            or is_mps_dispatch_key(k),
```

Comment:
Hey @albanD @malfet Could you please help review?

And should there be a device check in `_copy_from_and_resize()`? Thanks!

```diff
# We need this to be able to properly copy from a CPU to an XLA tensor with different sizes.
# See https://github.com/pytorch/xla/issues/2881
 - func: _copy_from_and_resize(Tensor self, Tensor dst) -> Tensor
+  device_check: NoCheck   # TensorIterator
   dispatch:
     MPS: _copy_from_and_resize_mps
   autogen: _copy_from_and_resize.out
```